ludwig/data/dataset_synthesizer.py

```diff
@@ -206,7 +217,12 @@ def generate_datapoint(features):
 
 
 def generate_category(feature):
-    return random.choice(feature["idx2str"])
+    if DECODER in feature:
```

Comment:
nit: should we be using the helper function that you defined above here: `_get_feature_encoder_or_decoder`